test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_enqueue.cpp

```diff
@@ -0,0 +1,563 @@
+        };
+
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        command_buffer = clCreateCommandBufferKHR(1, &queue, nullptr, &error);
+        test_error(error, "clCreateCommandBufferKHR failed");
+
+        error = RecordCommandBuffer();
+        test_error(error, "RecordCommandBuffer failed");
+        error = verify_state(CL_COMMAND_BUFFER_STATE_EXECUTABLE_KHR);
+        test_error(error, "State is not Executable");
+
+        error = EnqueueCommandBuffer();
+        test_error(error, "EnqueueCommandBuffer failed");
+        error = verify_state(CL_COMMAND_BUFFER_STATE_PENDING_KHR);
+        test_error(error, "State is not Pending");
+
+        return CL_SUCCESS;
+    }
+
+    cl_int RecordCommandBuffer()
+    {
+        cl_int error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 0, nullptr, nullptr, nullptr);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        return CL_SUCCESS;
+    }
+
+    cl_int EnqueueCommandBuffer()
+    {
+        cl_int pattern = 0xE;
+
+        cl_int error =
+            clEnqueueFillBuffer(queue, out_mem, &pattern, sizeof(cl_int), 0,
+                                data_size(), 0, nullptr, nullptr);
+        test_error(error, "clEnqueueFillBuffer failed");
+
+        user_event = clCreateUserEvent(context, &error);
+        test_error(error, "clCreateUserEvent failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 1,
+                                          &user_event, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        return CL_SUCCESS;
+    }
+    clEventWrapper user_event;
+};
+
+// CL_INVALID_VALUE if queues is NULL and num_queues is > 0, or queues is not
+// NULL and num_queues is 0.
+struct EnqueueCommandBufferNullQueuesNumQueues : public BasicCommandBufferTest
+{
+    EnqueueCommandBufferNullQueuesNumQueues(cl_device_id device,
+                                            cl_context context,
+                                            cl_command_queue queue)
+        : BasicCommandBufferTest(device, context, queue), queue1(nullptr),
+          queue2(nullptr)
+    {}
+
+    cl_int Run() override
+    {
+        cl_int error = clEnqueueCommandBufferKHR(1, nullptr, command_buffer, 0,
+                                                 nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clEnqueueCommandBufferKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        cl_command_queue queues[2] = { queue1, queue2 };
+        error = clEnqueueCommandBufferKHR(0, queues, command_buffer, 0, nullptr,
```

Comment:
Can we just test this with a single queue, as there's a separate error for more than 1 queue, which could get confused with this.

We also don't check "queues is NULL and num_queues is > 0" in this test.